packages/libs/src/sdk/api/challenges/ChallengesApi.ts

```diff
@@ -91,6 +91,20 @@ export class ChallengesApi extends GeneratedChallengesApi {
    * @see {@link generateSpecifier} to create the specifier argument.
    */
   public async claimReward(request: ClaimRewardsRequest) {
+    for (let i = 0; i < 5; i++) {
```

Comment:
@rickyrombo what do you think about adding retries on the whole thing for now? even w the fix below there are still some `block height exceeded` or `NotEnoughSigners` errors that eventually resolve for some reason.

Reply:
I want to solve these problems more generally and more deeply understand why they happen. I think this _should_ work but it means every time we interact with Solana in a pattern like this we have to do fairly aggressive client-side retries, which _usually_ is quite wasteful as we already have many, many retries on the relay side.

Before these changes, we _did_ have client side retries and the results weren't much better - we just failed over and over and the experience was actually longer for the user and still failed. Without understanding _why_ we get the error and blindly retrying, we may run into this again...

I think we should roll out your initial change w/o retries and see what happens in our analytics, and see if `NotEnoughSigners` errors go down

Reply:
yeaa i get you. just unfortunate that i think it'll still be unstable but let's see and sync if it's still an issue.

Reply:
Yeah the irony is sometimes these "failures" aren't actually real. Like "TransactionExpiredBlockheightExceededError" sometimes occurs even though the transaction actually does make it through. I wonder if there's some issue in the assumptions made by the `confirmTransaction` function? This one in particular has been boggling my brain. If the transaction expires, that means the blockhash is out of the list of onchain blockhashes that are valid, which means should the transaction now get processed, it will be rejected with "blockhashnotfound". So how are these transactions succeeding? Are the blockhashes still lingering around past their expiry? Is there a race condition with the expiry and the confirmation? What's going on??

Reply:
```
{"error":"TransactionExpiredBlockheightExceededError: Signature 5da55Xj5cvhRw8xZZMgMZfPZerFnfHW1hLgv2DsarhwC97LWKh56FGGLqbDvekzwechuZVPopisMfTioygLvZGun has expired: block height exceeded.","signature":"5da55Xj5cvhRw8xZZMgMZfPZerFnfHW1hLgv2DsarhwC97LWKh56FGGLqbDvekzwechuZVPopisMfTioygLvZGun"}
```
for example, from amplitude, succeeded: https://explorer.solana.com/tx/5da55Xj5cvhRw8xZZMgMZfPZerFnfHW1hLgv2DsarhwC97LWKh56FGGLqbDvekzwechuZVPopisMfTioygLvZGun